src/FilesystemSyncTaskHelper.php

```diff
@@ -56,20 +57,51 @@ class FilesystemSyncTaskHelper
      */
     private $delete_broken;
 
+    /**
+     * @var FlysystemAssetStore
+     */
+    private $flysystemAssetStore;
+
+    /**
+     * @var PublicAssetAdapter
+     */
+    private $publicAssetAdapter;
+
     /**
      * @param array $options
      * @return array An array containing results
      * @throws \SilverStripe\ORM\ValidationException
      */
     public function run(array $options)
+    {
+        $this->prepare($options);
+
+        // Check for db files with missing filesystem files and remove those db entries
+        $this->deleteBroken();
+
+        // Sync Files
+        $this->syncFiles();
+
+        // Remove Directories
+        $this->removeDirectories();
+
+        return $this->results;
+    }
+
+    /**
+     * Prepare for execution
+     * @param $options
```

Comment:
Missing type hint. If it's an array you can put the typehint in the function arguments too.